### PR TITLE
CI: Slack on merge build failure

### PR DESCRIPTION
**Pull request by gavin-stackrox**
## Description

In the unlikely event of a build failure on merge slack to #test-failures. This restores previous behavior from Circle CI days.

## Checklist
- [x] Investigated and inspected CI test results

## Testing Performed

- [x] Skipped for pull request - https://github.com/stackrox/stackrox/actions/runs/3834092599
- [x] Posts for push - https://github.com/stackrox/stackrox/actions/runs/3834088051 - https://srox.slack.com/archives/C740BUV25/p1672792532350339

### CODE REVIEW

**Review by janisz — APPROVED** [review]

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -484,3 +484,25 @@ jobs:
...
+      - build-and-push-main-rcd
+      - build-and-push-operator
+      - build-and-push-mock-grpc-server
+      - scan-images-with-roxctl
+    steps:
+      - name: Slack Workflow Notification
+        uses: Gamesight/slack-workflow-status@v1.1.0
+        with:
+          repo_token: ${{secrets.GITHUB_TOKEN}}
+          slack_webhook_url: ${{secrets.TEST_FAILURES_NOTIFY_WEBHOOK}}
```
Comment (tommartensen) [inline .github/workflows/build.yaml:508]: Your test alert went to #slack-test channel, do you plan to use a different one?

Comment (gavin-stackrox) [reply .github/workflows/build.yaml:508]: Yeah, this commit sends them to #test-failures https://github.com/stackrox/stackrox/pull/4272/commits/e87e3da256829b7efbe9258d638862e07d9f0975

Comment (tommartensen) [reply .github/workflows/build.yaml:508]: Perfect!

**Review by tommartensen — APPROVED** [review]
LGTM

**Pull request merged**